db/db_impl/db_impl.cc

```diff
@@ -4890,26 +4910,28 @@ Status DBImpl::VerifyChecksumInternal(const ReadOptions& read_options,
   return s;
 }
 
-Status DBImpl::VerifySstFileChecksum(const FileMetaData& fmeta,
+Status DBImpl::VerifySstFileChecksum(const std::string& file_checksum_expected,
```

Comment:
How about renaming this to `VerifyFullFileChecksum` to make it clear it's not solely for SST files?